cumulocity-blocks/TimeTicker.mon

```diff
@@ -95,13 +90,13 @@ event TimeTicker{
 	 */
 	action $init() {
 		string id;
-		for id in devices{
+		for id in inputHandler.devices{
 			try {
 				TimerParams tp := TimerParams.recurring($parameters.periodSecs);
```

Comment:
I think, you can use `partitionForTimer` action of the input handle to simplify it.
```
TimerParams tp := TimerParams.recurring($parameters.periodSecs).withPartition(inputHandler.partitionForTimer(id));
any discard := $base.createTimerWith(tp);
```

Reply:
I had to remove the reference to the devices as again there is no event type that I can declare in the input handler. What do you think?

Reply:
If you really want to use deviceId, you can either use one of the existing Cumulocity event like Operation or define a dummy event for this block only which has a source or id field. Of course, you'll not be really listening for that event.

Reply:
ok, I did this for the IntervalPulseGenerator.